soc/arm/st_stm32/stm32g4/soc.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2019 Richard Osterloh <richard.osterloh@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file SoC configuration macros for the STM32G4 family processors.
+ *
+ * Based on reference manual:
+ *   STM32G4xx advanced ARM ® -based 32-bit MCUs
+ *
+ * Chapter 2.2: Memory organization
+ */
+
+
+#ifndef _STM32G4_SOC_H_
+#define _STM32G4_SOC_H_
+
+#ifndef _ASMLANGUAGE
+
+#include <autoconf.h>
+#include <stm32g4xx.h>
+
+/* ARM CMSIS definitions must be included before kernel_includes.h.
+ * Therefore, it is essential to include kernel_includes.h after including
+ * core SOC-specific headers.
+ */
+#include <kernel_includes.h>
```

Comment:
Please, for new boards, follow the changed we have been introducing in patches such as #19357 , i.e. do not included kernel header files in soc.h and, instead, include auto-generated DTS headers

Reply:
@ioannisg Makes sense. Changed.